Source/WebCore/dom/TrustedTypePolicyFactory.cpp

```diff
@@ -100,21 +101,59 @@ Ref<TrustedScript> TrustedTypePolicyFactory::emptyScript() const
     return TrustedScript::create(""_s);
 }
 
-String TrustedTypePolicyFactory::getAttributeType(const String& tagName, const String& attribute, const String& elementNamespace, const String& attrNamespace) const
+String TrustedTypePolicyFactory::getAttributeType(const String& tagName, const String& attributeParam, const String& elementNamespace, const String& attrNamespace) const
 {
-    UNUSED_PARAM(tagName);
-    UNUSED_PARAM(attribute);
-    UNUSED_PARAM(elementNamespace);
-    UNUSED_PARAM(attrNamespace);
-    return String();
+    auto localName = tagName.convertToASCIILowercase();
+    auto attributeName = attributeParam.convertToASCIILowercase();
+
+    if (attributeName.startsWith("on"_s))
+        return trustedTypeToString(TrustedType::TrustedScript);
+
+    AtomString elementNS = elementNamespace.isEmpty() ? HTMLNames::xhtmlNamespaceURI : AtomString(elementNamespace);
+    AtomString attrNS = attrNamespace.isEmpty() ? nullAtom() : AtomString(attrNamespace);
+
+    const QualifiedName element(nullAtom(), AtomString(localName), elementNS);
+    const QualifiedName attribute(nullAtom(), AtomString(attributeName), attrNS);
```

Comment:
Not sure the `const` adds anything here. There are lots of local variables that we don’t change. I would leave it out.